app/Rules/StatusCode404.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Rules;
+
+use App\Services\UrlFetcher;
+use GuzzleHttp\Exception\BadResponseException;
+
+class StatusCode404 extends Rule
+{
+    protected $statusCode;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function check()
+    {
+        $uri = rtrim($this->url, '/').'/page-to-test-404-response';
```

Comment:
I made it a bit more unique; `/-~!page-to-test-404-responses-for-invalid-pages!~-`
But not that you don't have to do anything on your server for this, it should work as long as the page doesn't actually exist.